lib/failure.sh

```diff
@@ -27,6 +27,20 @@ fail_invalid_package_json() {
   fi
 }
 
+fail_dot_heroku_node() {
+  if [ -f "${1:-}/.heroku/node" ]; then
```

Comment:
> this will only catch node as a file; what if it's a directory?

This happens if the buildpack runs twice for some reason, so I'm going to leave it as just a file check